### PR TITLE
Revamp dark mode styling throughout the app 🌚

### PR DESCRIPTION
**Pull request by madsnedergaard**
## Description

The dark mode styling just got a nice revisit where @rasmusjeppsen and I tweaked colors and general styling to make it all more consistent and nice.

### Preview

Preview branch here: https://b31cae97.electricitymaps-staging.pages.dev/

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
Looks like all the shades are a little lighter (looks good though) but just double checking that you guys are aware of the new 950 shade since tailwind 3.3?

File: web/src/components/ToggleButton.tsx [hunk 1]
```diff
@@ -41,13 +41,13 @@ export default function ToggleButton({
   return (
     <div
       className={twMerge(
-        'z-10 flex h-9 rounded-full bg-gray-200/80 px-1  shadow    dark:bg-gray-900',
-        transparentBackground ? 'bg-gray-200/80 backdrop-blur-sm' : 'bg-gray-200'
+        'z-10 flex h-9 rounded-full bg-gray-200/80 px-1  shadow    dark:bg-gray-800/80',
```
Comment (VIKTORVAV99) [inline web/src/components/ToggleButton.tsx:44]:
```suggestion
        'z-10 flex h-9 rounded-full bg-gray-200/80 px-1 shadow dark:bg-gray-800/80',
```

File: web/src/components/ToggleButton.tsx [hunk 2]
```diff
@@ -81,7 +81,7 @@ export default function ToggleButton({
                 role="button"
                 tabIndex={0}
                 className={twMerge(
-                  'ml-2 inline-flex h-[29px] w-[29px] select-none justify-center self-center rounded-full bg-white shadow dark:border dark:border-gray-500 dark:bg-gray-900',
+                  'ml-2 inline-flex h-[29px] w-[29px] select-none justify-center self-center rounded-full bg-white shadow  dark:border dark:border-gray-500/80 dark:bg-gray-600/50',
```
Comment (VIKTORVAV99) [inline web/src/components/ToggleButton.tsx:84]:
```suggestion
                  'ml-2 inline-flex h-[29px] w-[29px] select-none justify-center self-center rounded-full bg-white shadow dark:border dark:border-gray-500/80 dark:bg-gray-600/50',
```

File: web/src/components/legend/LegendContainer.tsx [hunk 3]
```diff
@@ -21,7 +21,7 @@ export default function LegendContainer(): ReactElement {
     windLayerToggle === ToggleOptions.ON && selectedDatetime.index === 24;
 
   return (
-    <div className="invisible fixed bottom-4 right-4 z-20 flex w-[224px]  flex-col rounded bg-white/90 px-1 py-2 shadow-xl backdrop-blur-sm dark:bg-gray-900 sm:visible ">
+    <div className="invisible fixed bottom-4 right-4 z-20 flex w-[224px]  flex-col rounded bg-white/90 px-1 py-2 shadow-xl backdrop-blur-sm dark:bg-gray-800 sm:visible ">
```
Comment (VIKTORVAV99) [inline web/src/components/legend/LegendContainer.tsx:24]:
```suggestion
    <div className="invisible fixed bottom-4 right-4 z-20 flex w-[224px] flex-col rounded bg-white/90 px-1 py-2 shadow-xl backdrop-blur-sm dark:bg-gray-800 sm:visible">
```

File: web/src/features/charts/tooltips/CarbonChartTooltip.tsx [hunk 4]
```diff
@@ -28,23 +28,14 @@ export default function CarbonChartTooltip({ zoneDetail }: InnerAreaGraphTooltip
   return (
     <div
       data-test-id="carbon-chart-tooltip"
-      className="w-full rounded-md bg-white p-3 shadow-xl dark:bg-gray-900 sm:w-80"
+      className="w-full rounded-md bg-white p-3 shadow-xl dark:border dark:border-gray-700 dark:bg-gray-800   sm:w-80"
```
Comment (VIKTORVAV99) [inline web/src/features/charts/tooltips/CarbonChartTooltip.tsx:31]:
```suggestion
      className="w-full rounded-md bg-white p-3 shadow-xl dark:border dark:border-gray-700 dark:bg-gray-800 sm:w-80"
```

File: web/src/features/charts/tooltips/EmissionChartTooltip.tsx [hunk 5]
```diff
@@ -19,7 +19,7 @@ export default function EmissionChartTooltip({ zoneDetail }: InnerAreaGraphToolt
   const { stateDatetime } = zoneDetail;
 
   return (
-    <div className="w-full rounded-md bg-white p-3 shadow-xl dark:bg-gray-900 sm:w-[350px]">
+    <div className="w-full rounded-md bg-white p-3 shadow-xl  dark:border dark:border-gray-700 dark:bg-gray-800  sm:w-[350px]">
```
Comment (VIKTORVAV99) [inline web/src/features/charts/tooltips/EmissionChartTooltip.tsx:22]:
```suggestion
    <div className="w-full rounded-md bg-white p-3 shadow-xl dark:border dark:border-gray-700 dark:bg-gray-800 sm:w-[350px]">
```

File: web/src/features/panels/faq/FAQContent.tsx [hunk 6]
```diff
@@ -62,7 +62,7 @@ function FAQContent() {
               </Accordion.Header>
               <Accordion.Content className="ml-2 border-l border-gray-300 pl-4 radix-state-closed:animate-slide-up radix-state-open:animate-slide-down">
                 <div
-                  className="prose prose-sm text-md leading-5 dark:prose-invert"
+                  className="prose prose-sm text-md leading-5 dark:prose-invert prose-a:text-sky-600 prose-a:no-underline  hover:prose-a:underline dark:prose-a:invert "
```
Comment (VIKTORVAV99) [inline web/src/features/panels/faq/FAQContent.tsx:65]:
```suggestion
                  className="prose prose-sm text-md leading-5 dark:prose-invert prose-a:text-sky-600 prose-a:no-underline hover:prose-a:underline dark:prose-a:invert "
```

File: web/src/features/panels/ranking-panel/InfoText.tsx [hunk 7]
```diff
@@ -14,7 +14,7 @@ function ExternalLink({ href, text }: { href: string; text: string }) {
 export default function InfoText() {
   const { __ } = useTranslation();
   return (
-    <div className="prose text-sm dark:prose-invert prose-p:my-1 prose-p:leading-snug prose-a:text-sky-600 prose-a:no-underline hover:prose-a:underline">
+    <div className="prose text-sm dark:prose-invert prose-p:my-1 prose-p:leading-snug prose-a:text-sky-600 prose-a:no-underline hover:prose-a:underline  dark:prose-a:invert">
```
Comment (VIKTORVAV99) [inline web/src/features/panels/ranking-panel/InfoText.tsx:17]:
```suggestion
    <div className="prose text-sm dark:prose-invert prose-p:my-1 prose-p:leading-snug prose-a:text-sky-600 prose-a:no-underline hover:prose-a:underline dark:prose-a:invert">
```

File: web/src/features/weather-layers/wind-layer/windy.ts [hunk 8]
```diff
@@ -437,7 +437,7 @@ var Windy = function (params: any) {
       return [];
     });
 
-    const particleCount = Math.round(bounds.width * PARTICLE_MULTIPLIER * zoomScaling());
+    let particleCount = Math.round(bounds.width * PARTICLE_MULTIPLIER * zoomScaling());
```
Comment (VIKTORVAV99) [inline web/src/features/weather-layers/wind-layer/windy.ts:440]: I was going to do this as well but didn't really want to do it in a unrelated PR...

**Review by VIKTORVAV99 — COMMENTED** [review]
I really like the transparency added here:
image

Is this something we can do for the icon buttons, co2ledgend and timeslider/selector as well?

File: web/src/features/weather-layers/wind-layer/windy.ts (hunk 8 above)
Comment (VIKTORVAV99) [reply web/src/features/weather-layers/wind-layer/windy.ts:440]: > I was going to do this as well but didn't really want to do it in a unrelated PR... 😅

Comment (madsnedergaard) [reply web/src/features/weather-layers/wind-layer/windy.ts:440]: Haha yeah, just seemed kinda silly to make a new PR to fix this ¯\\\_(ツ)\_/¯

**Comment by madsnedergaard** [comment]
> I really like the transparency added here: image
> 
> Is this something we can do for the icon buttons, co2legend and timeslider/selector as well?

Yeah, I would love to do that afterwards, but figured it deserved it's own PR :)

**Comment by madsnedergaard** [comment]
> Looks like all the shades are a little lighter (looks good though) but just double checking that you guys are aware of the new 950 shade since tailwind 3.3?

Didn't actually know that, but we also decided that it's better to go with lighter to increase contrasts - going with 900, 950 and black means very low contrast in most cases :)

Also thanks for fixing all the spaces, so annoying the Prettier plugin doesn't do it 😅

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Comment by tonypls** [comment]
I think the legend bg would look good if it matched the tooltip / leftpanel bg. Kinda follows a pattern of if it's map -> bg it's one one color then we go lighter for layers on top. 

image

**Comment by tonypls** [comment]
> Hmm yeah, it just seemed a bit weird that the timecontroller-box and legend-box weren't of the same color as they both "float" on top of things :) How about us merging this now, and then syncing with @rasmusjeppsen next week?

Sounds good to me

**Review by tonypls — APPROVED** [review]
🟢

**Pull request merged**